client/src/components/profile/ProfileForm.vue

```diff
@@ -1,9 +1,216 @@
+                        :items="timeZones"
+                        :label="lang.TIME_ZONE"
+                        solo
+                        outlined
+                        required
+                    ></VSelect>
+                </VCol>
+                <VCol cols="12">
+                    <VSpacer></VSpacer>
+                    <VSpacer></VSpacer>
+                </VCol>
+                <VAlert cols="12" type="error" v-if="errorMessage">
+                    {{ errorMessage }}
+                </VAlert>
+                <VCol>
+                    <div>
+                        <VBtn class="ma2" @click="resetChanges">
+                            {{ lang.CANCEL }}
+                        </VBtn>
+                        <VBtn
+                            class="ma-2"
+                            color="primary"
+                            dark
+                            @click="onSaveHandle"
+                            >{{ lang.SAVE }}
+                        </VBtn>
+                    </div>
+                </VCol>
+            </VRow>
+        </VContainer>
+    </VForm>
 </template>
 
 <script>
+import enLang from '@/store/modules/i18n/en';
+import { mapActions, mapGetters } from 'vuex';
+
 export default {
-    name: 'ProfileForm'
+    name: 'ProfileForm',
+    data: () => ({
+        lang: enLang,
+        file: null,
+        newAvatar: null,
+        errorMessage: '',
+        userProfile: {
+            name: '',
+            welcome: '',
+            language: enLang.ENGLISH,
+            dateFormat: 'rcf-1123',
+            timeFormat: '12h',
+            country: '',
+            timeZone: null
+        },
+
+        languages: [enLang.ENGLISH, enLang.GERMAN, enLang.UKRAINIAN],
+        dateFormats: [
+            { value: 'rcf-1123', text: 'Mon, 10 Aug 2020 14:38:15 +0000' },
+            { utc: 'utc', text: '2020-08-10T14:38:15Z' },
+            { value: 'iso-8601', text: '2020-08-10T14:38:15+0000' }
+        ],
+        timeFormats: ['12h', '24h'],
+        timeZones: []
+    }),
+
+    created() {
+        this.userProfile = {
+            ...this.user
+        };
+
+        this.newAvatar = this.avatar;
+    },
+
+    computed: {
+        ...mapGetters('profile', {
+            avatar: 'getAvatar'
+        }),
+
+        ...mapGetters('auth', {
+            user: 'getLoggedUser'
+        }),
+
+        avatarIsNew() {
+            return this.avatar === this.newAvatar;
+        }
+    },
+
+    methods: {
+        ...mapActions('profile', ['updateAvatar', 'updateProfile']),
+
+        updateImage(event) {
+            this.file = event.target.files[0];
+            this.newAvatar = URL.createObjectURL(this.file);
+        },
+
+        async onSaveHandle() {
+            try {
+                if (this.avatarIsNew) {
+                    const response = await this.updateAvatar(this.avatar);
+                    const avatar = response?.avatar?.url;
+                    this.userProfile = avatar;
```

Comment:
why do you replace whole userProfile with avatar?

Reply:
flow must be, to upload file, then update whole profile and change global state.